modules/integration/tests-common/admin-clients/src/main/java/org/wso2/mb/integration/common/clients/AndesJMSConsumer.java

```diff
@@ -0,0 +1,576 @@
+            log.error("Error while listening to messages", e);
+            throw new RuntimeException("JMSException : Error while listening to messages", e);
+        } catch (IOException e) {
+            log.error("Error while writing message to file", e);
+            throw new RuntimeException("IOException : Error while writing message to file\"", e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void onMessage(Message message) {
+        try {
+            this.processReceivedMessage(message);
+        } catch (JMSException e) {
+            log.error("Error while listening to messages", e);
+            throw new RuntimeException("Error while listening to messages", e);
+        } catch (IOException e) {
+            log.error("Error while writing message to file", e);
+            throw new RuntimeException("Error while listening to messages", e);
+        }
+    }
+
+    /**
+     * Processes the received messages. The processing includes the following actions. 1.
+     * Calculation of transactions per second. 2. Calculation of  average latency for messages. 3.
+     * Message detail logging 4. Writes messages to a file. 5. Writes statistics to a file. 6.
+     * Closing and un-subscribing of client.
+     *
+     * @param message The {@link javax.jms.Message} to publish.
+     * @return true if client is stopped or un-subscribed, false otherwise.
+     * @throws JMSException
+     * @throws IOException
+     */
+    private boolean processReceivedMessage(Message message)
+            throws JMSException, IOException {
+        if (null != message) {
+            long threadID = Thread.currentThread().getId();
+            // Calculating total latency
+            long currentTimeStamp = System.currentTimeMillis();
+            this.totalLatency = this.totalLatency + (currentTimeStamp - message.getJMSTimestamp());
+            // Setting timestamps for TPS calculation
+            if (0 == this.firstMessageConsumedTimestamp) {
+                this.firstMessageConsumedTimestamp = currentTimeStamp;
+            }
+            this.lastMessageConsumedTimestamp = currentTimeStamp;
+
+            if (message instanceof TextMessage) {
+                this.receivedMessageCount.incrementAndGet();
+                TextMessage textMessage = (TextMessage) message;
+                JMSDeliveryStatus deliveryStatus;
+                // Gets whether the message is original or redelivered
+                if (message.getJMSRedelivered()) {
+                    deliveryStatus = JMSDeliveryStatus.REDELIVERED;
+                } else {
+                    deliveryStatus = JMSDeliveryStatus.ORIGINAL;
+                }
+                // Logging the received message
+                if (0 == this.receivedMessageCount.get() % this.consumerConfig
+                        .getPrintsPerMessageCount()) {
+                    log.info("[RECEIVE] ThreadID:" + threadID + " Destination(" + this.consumerConfig
```

Comment:
The implemented client which uses the following method is used by MB test cases in product-mb. When running the test cases is is vital for us to know that all messages are properly sent or received. Therefore log.info has been used.